### ISSUE TITLE
difficult problem

### ISSUE THREAD

**Issue by not-a-sysadmin** [ibus/ibus#2702]
(stand-alone workstation; Fedora-40; gnome with X11; IBus 1.5.30)

(this was initially sent to a Redhat ibus expert who referred me to this github page)

I've been using ibus for input of Chinese characters via pinyin for over 11 years.  I upgraded this stand-alone home workstation from Fedora-39 to Fedora-40 early last month.  Since then, I have not been able to use ibus.  super-space and control-space do nothing.  There is no hint of an ibus icon along the top of the display, nothing when I click the corner icon (looks something like a button or clock at noon), nothing in gnome settings. Chinese characters do display properly in Firefox (sometimes), files with Chinese characters, and LibreOffice products containing Chinese characters.

I thought about removing (dnf remove) and then installing (dnf install) ibus.  But when I enter "dnf remove", the list of packages includes packages that seem to me should not be removed:
- - - - - -
-bash.1[~]: dnf --refresh remove ibus
Dependencies resolved.
================================================================================
 Package                         Arch   Version Repository     Size
================================================================================
Removing:
 ibus                            x86_64 1.5.30-7.fc40 @updates      141 M
Removing dependent packages:
 gdm                             x86_64 1:46.2-1.fc40 @updates      5.4 M
 gnome-shell                     x86_64 46.6-2.fc40 @updates       14 M
 gnome-shell-extension-ibus-font noarch 0.20230705-4.fc40 @fedora        21 k
 ibus-qt                         x86_64 1.3.4-11.fc40 @fedora       440 k
 zoom                            x86_64 6.2.6.2503-1 @@commandline 682 M
Removing unused dependencies:
 colord-gtk4                     x86_64 0.3.1-1.fc40 @fedora        36 k
 evolution-data-server           x86_64 3.52.4-1.fc40 @updates      8.7 M
 evolution-data-server-langpacks noarch 3.52.4-1.fc40 @updates      8.7 M
 gnome-control-center            x86_64 46.5-1.fc40 @updates       24 M
 gnome-session-wayland-session   x86_64 46.0-1.fc40 @fedora        16 k
 gnome-session-xsession          x86_64 46.0-1.fc40 @fedora        16 k
 gnome-tour                      x86_64 46.0-2.fc40 @updates      2.3 M
 ibus-m17n                       x86_64 1.4.34-1.fc40 @updates      325 k
 ibus-panel                      x86_64 1.5.30-7.fc40 @updates      282 k
 ibus-setup                      noarch 1.5.30-7.fc40 @updates      327 k
 libphonenumber                  x86_64 8.13.35-1.fc40 @updates      6.8 M
 malcontent                      x86_64 0.12.0-1.fc40 @updates      520 k
 malcontent-control              x86_64 0.12.0-1.fc40 @updates      122 k
 malcontent-ui-libs              x86_64 0.12.0-1.fc40 @updates      117 k
 pipewire-gstreamer              x86_64 1.0.9-1.fc40 @updates      133 k
 power-profiles-daemon           x86_64 0.23-1.fc40 @updates      183 k
 python3-pam                     noarch 2.0.2-8.fc40 @fedora        58 k
 qadwaitadecorations-qt5         x86_64 0.1.5-7.fc40 @updates      152 k
 qadwaitadecorations-qt6         x86_64 0.1.5-7.fc40 @updates      174 k
 tecla                           x86_64 45.0-2.fc40 @fedora       119 k
 xdg-desktop-portal-gnome        x86_64 46.2-1.fc40 @updates      844 k

Transaction Summary
================================================================================
Remove  27 Packages

Freed space: 896 M
Is this ok [y/N]: N
Operation aborted.
-bash.2[~]:
- - - - - -
By the way, "dnf --refresh upgrade ibus" results in "Nothing to do."

How do I get ibus working properly?  For multiple reasons (some private), I cannot do a total re-install, and I cannot at this time upgrade to Fedora 41.

By the way, I am a home user; I do not have systems administration training or experience.  Also, although this workstation is at Fedora-40, there are indications that it is using dnf5, not dnf4.

(from the Redhat employee's private respone)
> Maybe you can try to add "export GTK_IM_MODULE=ibus" into ~/.bash_profile.

I moved the following 3 lines from my .bashrc to my .bash_profile:
export GTK_IM_MODULE=ibus
export XMODIFIERS=@im=ibus
export QT_IM_MODULE=ibus
I then rebooted.  No change.

> And add some input method in gnome-control-center.

Result:
bash.13[~]: gnome-control-center
11:45:57.2286                  Gtk[ 7895]: WARNING: No IM module matching GTK_IM_MODULE=ibus found

I saw no way of adding anything to the gnome-control-center, even after checking the GUI's "help".
How do I add the ibus to the gnome-control-center?

**Comment by fujiwarat** [comment]
You should use the bug templete and you didn't mention some important info.

> (stand-alone workstation; Fedora-40; gnome with X11; IBus 1.5.30)

Do you use the GNOME Xorg desktop environment? You can check with `env | grep XDG_SESSION_TYPE` command.

> , I have not been able to use ibus. super-space and control-space do nothing

Which application did you try with?

> bash.13[~]: gnome-control-center
> 11:45:57.2286 Gtk[ 7895]: WARNING: No IM module matching GTK_IM_MODULE=ibus found

Did you install ibus packages? `rpm -qa | grep ibus`

**Comment by not-a-sysadmin** [comment]
```
> You should use the bug templete and you didn't mention some important info.
> 
> > (stand-alone workstation; Fedora-40; gnome with X11; IBus 1.5.30)

bash.2[~]: env | grep XDG_SESSION_TYPE
XDG_SESSION_TYPE=x11
bash.3[~]:

> Do you use the GNOME Xorg desktop environment? You can check with `env | grep XDG_SESSION_TYPE` command.
> 
> > , I have not been able to use ibus. super-space and control-space do nothing
> 
> Which application did you try with?

gnome terminal, gvim, LibreOffice, Thunderbird, Firefox. maybe others.   I've found nothing in which ibus does work or is available.

I've attached a screen-capture showing the top of the screen. Before upgrading from Fedora-39 to Fedora-40 (in early October), there was an "en" showing at the top of the screen, just to the left of the internet icon.  Ibus was easy to access and use.

> > bash.13[~]: gnome-control-center
> > 11:45:57.2286 Gtk[ 7895]: WARNING: No IM module matching GTK_IM_MODULE=ibus found
> 
> Did you install ibus packages? `rpm -qa | grep ibus`

I installed it 11 years ago.  Using yum and dnf, I have kept it up-to-date since then.

bash.1[~]: rpm -qa | grep ibus
libusal-1.1.11-54.fc40.x86_64
libusb-compat-0.1-0.1.8-7.fc40.x86_64
libusbmuxd-2.0.2^20230620git8d30a55-5.fc40.x86_64
libusbmuxd-2.0.2^20230620git8d30a55-5.fc40.i686
ibus-qt-1.3.4-11.fc40.x86_64
scribus-1.6.1-3.fc40.x86_64
libuser-0.64-8.fc40.x86_64
ibus-m17n-1.4.34-1.fc40.x86_64
ibus-libs-1.5.30-7.fc40.x86_64
ibus-gtk2-1.5.30-7.fc40.x86_64
ibus-gtk3-1.5.30-7.fc40.x86_64
ibus-panel-1.5.30-7.fc40.x86_64
ibus-setup-1.5.30-7.fc40.noarch
ibus-1.5.30-7.fc40.x86_64
ibus-wayland-1.5.30-7.fc40.x86_64
libusb1-1.0.27-4.fc40.x86_64
libusb1-1.0.27-4.fc40.i686
gnome-shell-extension-ibus-font-0.20230705-4.fc40.noarch
-bash.2[~]:

I've also checked what's actually running:

bash.15[Downloads]: ps -ef | grep ibus
weilian+    2486    1964  0 07:29 ?        00:00:05 /usr/bin/ibus-daemon --panel disable --xim
weilian+    2741    2486  0 07:29 ?        00:00:00 /usr/libexec/ibus-dconf
weilian+    2742    2486  0 07:29 ?        00:00:03 /usr/libexec/ibus-extension-gtk3
weilian+    2768    1964  0 07:29 ?        00:00:00 /usr/libexec/ibus-x11 --kill-daemon
weilian+    2770    1964  0 07:29 ?        00:00:00 /usr/libexec/ibus-portal
weilian+    2873    2486  0 07:30 ?        00:00:01 /usr/libexec/ibus-engine-simple
weilian+  134403  130773  0 11:28 pts/0    00:00:00 grep --color=auto ibus
bash.16[Downloads]:

Important notes:
Either the upgrade to Fedora-40 or its post-upgrade step "remove-retired-packages" seem to have deleted many things it should not have, and/or corrupted database(s?) used by dnf.  The dnf remove output in my original post illustrates that.  I am not a sys.admin., so I do not know how to dig into those things.  I do not think ibus itself has a bug.  I think the problem is on my workstation.  I need your help to fix ibus on my workstation.  For the rest, I'll be going back to the Fedora user's list.

Is there anything else I can provide to you to help?  screen-capture of gnome settings gui or the tweaks tool?

```
![no_ibus](https://github.com/user-attachments/assets/bfeb5b32-e0ef-4c08-95bc-d7ef46e44bf8)

**Comment by fujiwarat** [comment]
Try to get the output if `ibus im-module --type gtk3` and `strace /usr/lib64/firefox/firefox |& grep immodule` after you install strace package.

**Comment by not-a-sysadmin** [comment]
> Try to get the output if `ibus im-module --type gtk3` and `strace /usr/lib64/firefox/firefox |& grep immodule` after you install strace package.
```

If it matters, I did this immediately after booting, logging in as a regular user, and launching a gnome terminal.  I shut down Firefox immediately after the home page loaded, and then copied and pasted this...
- - -
bash.1[~]: ibus im-module --type gtk3
ibus
bash.2[~]: strace /usr/lib64/firefox/firefox |& grep immodule
openat(AT_FDCWD, "/usr/lib64/gtk-3.0/3.0.0/immodules.cache", O_RDONLY) = 37
newfstatat(AT_FDCWD, "/usr/lib64/gtk-3.0/3.0.0/immodules/im-ibus.so", {st_mode=S_IFREG|0755, st_size=40992, ...}, 0) = 0
openat(AT_FDCWD, "/usr/lib64/gtk-3.0/3.0.0/immodules/im-ibus.so", O_RDONLY|O_CLOEXEC) = 37
bash.3[~]:
- - -
Thank-you, Takao.

```

**Comment by fujiwarat** [comment]
From your strace result, I think ibus is configured properly.

> How do I add the ibus to the gnome-control-center?
> I installed it 11 years ago. 

Why haven't you known it for 11 years. Anyway,
After you install "ibus-hangul" package, you could try to run `gnome-control-center keyboard` -> Click "Add Input Source" button in "Input Sources" section -> Click "⫶"  and show more languages in "Add Input Source" dialog -> Click "Korean" -> "Select "Korean (Hangul)" and click "Add" button.
Basically IBus engines shows a configure icon in the row of the dialog box.

I think `strace /usr/lib64/firefox/firefox |& grep immodule` shows the same result above and I think you can use the Hangul typing when you select Hangul icon from the IBus panel icon memu?

`gsettings get org.gnome.desktop.input-sources sources` command shows "('ibus', 'hangul')" element to indicate an IBus engine.

**Comment by not-a-sysadmin** [comment]
```
> From your strace result, I think ibus is configured properly.
> 
> > How do I add the ibus to the gnome-control-center?
> > I installed it 11 years ago.
> 
> Why haven't you known it for 11 years.

well, after 11 years, I forgot.  :(

> Anyway, After you install "ibus-hangul" package, you could try to run
> `gnome-control-center keyboard` -> Click "Add Input Source" button in "Input Sources"
> section -> Click "⫶" and show more languages in "Add Input Source" dialog -> Click
> "Korean" -> "Select "Korean (Hangul)" and click "Add" button. Basically IBus engines
> shows a configure icon in the row of the dialog box.

done.
I think it works, but I don't know Korean, so I'm not sure!

> I think `strace /usr/lib64/firefox/firefox |& grep immodule` shows the same result
> above and I think you can use the Hangul typing when you select Hangul icon from
> the IBus panel icon memu?
> 
> `gsettings get org.gnome.desktop.input-sources sources` command shows
> "('ibus', 'hangul')" element to indicate an IBus engine.

bash.2[~]: gsettings get org.gnome.desktop.input-sources sources
[('xkb', 'us'), ('xkb', 'cn'), ('ibus', 'libpinyin'), ('ibus', 'hangul')]
bash.3[~]: 

My apologies for not being more specific.  I'm wanting to use pinyin to enter "simplified"
Chinese and "traditional" Chinese.  I got that working 11 years ago.  About 2 months ago, my
upgrade from Fedora-39 to Fedora-40 broke some things, including ibus.  After doing the
Korean steps that you provided above, I re-checked and found that ibus-libpinyin is missing.
I re-installed it.  I can now enter "simplified" Chinese.  I'll have to hunt down some
"traditional" Chinese to test that.

谢谢, Takao.

```

**Comment by fujiwarat** [comment]
Closing as not an ibus core.
If you still need helps. you could ask Fedora i18n mail or chats.
